src/data/finmap.lean

```diff
@@ -55,6 +55,27 @@ end
 @[simp] theorem lift_on_to_finmap {γ} (s : alist β) (f : alist β → γ) (H) :
   lift_on ⟦s⟧ f H = f s := by cases s; refl
 
+/-- Lift a permutation-respecting function on 2 `alist`s to 2 `finmap`s. -/
+@[elab_as_eliminator] def lift_on₂
+  {γ} (s₁ s₂ : finmap β) (f : alist β → alist β → γ)
+  (H : ∀ a₁ b₁ a₂ b₂ : alist β, a₁.entries ~ a₂.entries → b₁.entries ~ b₂.entries → f a₁ b₁ = f a₂ b₂) : γ :=
```

Comment:
Is there a reason this can't be built using iterated `lift_on`? Seems like it would avoid having to do the roption thing all over again.

Reply:
No reason. This was mainly trying to bring my own `lift_on₂` up to date with the existing `lift_on`. You could probably make the change much faster than I, so would you mind?

Reply:
I'm a bit busy at the moment, but if I have time later I can get to this

Reply:
Thanks! No hurry. I'm also working on other stuff.

Reply:
Done in 731dde4.